tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -107,10 +107,6 @@ public static void Initialize()
                 Log.Debug("Sending CallTarget integration definitions to native library.");
                 var payload = InstrumentationDefinitions.GetAllDefinitions();
                 NativeMethods.InitializeProfiler(payload.DefinitionsId, payload.Definitions);
-                foreach (var def in payload.Definitions)
-                {
-                    def.Dispose();
-                }
```

Comment:
don't understand why these disposes are no longer necessary? Do we want to keep the definitions in memory for the life time of the app, rather than recreate them each time they're used?

Reply:
This was causing problems when trying to uninstrument and reinstrument certain `CallTargets`. That's why I've moved it to the end in
`        private static void RunShutdown()
        {
            InstrumentationDefinitions.Dispose();
        }
`

Reply:
The problem is CallTargetDefinitions are created by the code gen and stored in a dictionary, not recreated each time.